workflows/cloudify_system_workflows/snapshots/credentials.py

```diff
@@ -13,52 +13,53 @@
 #    * See the License for the specific language governing permissions and
 #    * limitations under the License.
 
+import itertools
 import os
-import shutil
 import pickle
+import re
+import shutil
+import string
+import subprocess
 
+from cloudify.manager import get_rest_client
 from cloudify.workflows import ctx
-from cloudify.exceptions import NonRecoverableError
 
-from .utils import is_compute, copy as copy_file
+from .constants import SECRET_STORE_AGENT_KEY_PREFIX
+from .utils import is_compute
+
+
+ALLOWED_KEY_CHARS = string.ascii_letters + string.digits + '-._'
+CRED_DIR = 'snapshot-credentials'
+DEPLOYMENTS_QUERY = """
+    SELECT nodes.id, deployments.id, properties
+    FROM nodes
+    JOIN deployments
+    ON nodes._deployment_fk = deployments._storage_id
+    JOIN tenants
+    ON deployments._tenant_id = tenants.id
+    WHERE tenants.name = %(tenant)s
+    ;
+"""
 
 
 class Credentials(object):
-    _CRED_DIR = 'snapshot-credentials'
     _CRED_KEY_NAME = 'agent_key'
 
-    def restore(self, tempdir, postgres):
-        self._postgres = postgres
-        dump_cred_dir = os.path.join(tempdir, self._CRED_DIR)
-        if not os.path.isdir(dump_cred_dir):
-            ctx.logger.info('Missing credentials dir: '
-                            '{0}'.format(dump_cred_dir))
-            return
-        restored_cred_dir = self._create_restored_cred_dir()
-        agent_key_path_dict = self._create_agent_key_path_dict()
-
-        for dep_node_id in os.listdir(dump_cred_dir):
-            self._restore_agent_credentials(
-                dep_node_id,
-                dump_cred_dir,
-                restored_cred_dir,
-                agent_key_path_dict
-            )
-
     def dump(self, tempdir):
-        archive_cred_path = os.path.join(tempdir, self._CRED_DIR)
+        archive_cred_path = os.path.join(tempdir, CRED_DIR)
         ctx.logger.debug('Dumping credentials data, '
                          'archive_cred_path: {0}'.format(archive_cred_path))
         os.makedirs(archive_cred_path)
 
         for deployment_id, n in self._get_hosts():
             props = n.properties
-            if 'cloudify_agent' in props and 'key' in props['cloudify_agent']:
+            agent_config = get_agent_config(props)
+            if 'key' in agent_config:
                 node_id = deployment_id + '_' + n.id
-                agent_key_path = props['cloudify_agent']['key']
+                agent_key = agent_config['key']
                 self._dump_agent_key(
                     node_id,
-                    agent_key_path,
+                    agent_key,
```

Comment:
What is this is not a key path?
Possible values:
1. Private key content.
2. Private key path.
3. get_secret intrinsic function.